src/drawtypes/iconset.cpp

```diff
@@ -20,9 +21,11 @@ namespace drawtypes {
 
     // If fuzzy matching is turned on, try that first before returning the fallback.
     if (fuzzy_match) {
+      // works by finding the *longest* matching icon to the given workspace id
+      int max_size = -1;
       for (auto const& icon : m_icons) {
         if (id.find(icon.first) != std::string::npos) {
-          return icon.second;
+          max_size = std::max(max_size, (int)icon.first.length());
```

Comment:
You're finding the length of the longest match, but you're not actually doing anything with it.

Reply:
You can also write some tests in `tests/unit_tests/drawtypes/iconset.cpp` to check that the function actually does what you want it to do ([testing documentation](https://polybar.readthedocs.io/en/latest/dev/testing.html))